### PR TITLE
[Feature 36] connects Profile screen and Filters screen to the Navigation Drawer

### LINKED ISSUE #44: [Feature 36] Connect Profile screen and Filters screen to the Navigation Drawer

### PR DESCRIPTION
**Pull request by jenniferjinyoungyang**
What:
connects the edit profile screen and the edit filters screen to the navigation drawer created by https://github.com/CSC207-2023Y-UofT/course-project-tindar/pull/40

Why:
Users can navigate to the edit profile screen and to the edit filters screen through the navigation drawer.

Implementation Details

Profile screen and filters screen now use fragments instead of activities.
The navigation feature uses one activity with multiple fragments to navigate to. So profile screen and filters screen implementation had to be converted to fragments to fit into this framework.

Created a shared view model
The blank nav activity (where the edit profile fragment and edit filters fragment exist) holds a user id of the logged-in user. This user id is passed from the main activity. The user id has to be shared among the activity and fragments under the activity so each fragment can use the user id and pull the data about the logged-in user to support its own functionality.

### CODE REVIEW

**Comment by jenniferjinyoungyang** [comment]
I still need to add more documentation - I created an issue to add comments / Javadocs for the edit profile and edit filters implementation. I will add it later in that issue.

File: app/src/main/java/com/courseproject/tindar/BlankNavActivity.java [hunk 1]
```diff
@@ -17,31 +18,32 @@
...
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
 
-        binding = ActivityBlankNavBinding.inflate(getLayoutInflater());
+        // retrieves user Id passed from other activity
+        Intent intent = getIntent();
+        String userId = intent.getStringExtra("user_id");
+
+        // instantiates the blank nav view model and sets the user id that is currently logged in
+        BlankNavViewModel blankNavViewModel = new ViewModelProvider(this).get(BlankNavViewModel.class);
```
Comment (navarhontes) [inline app/src/main/java/com/courseproject/tindar/BlankNavActivity.java:32]: Do we have to make a new one here, or is it possible to get it passed in from the previous screen?

File: app/src/main/java/com/courseproject/tindar/BlankNavViewModel.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * A simple {@link ViewModel} subclass. This is a view model for the blank nav activity to share data among the activity
+ * and its fragments.
+ */
+public class BlankNavViewModel extends ViewModel {
+
+    /**
+     * user id for the logged in user
+     */
+    private final MutableLiveData<String> userId = new MutableLiveData<>();
```
Comment (navarhontes) [inline app/src/main/java/com/courseproject/tindar/BlankNavViewModel.java:16]: Why is this mutable? I'm currently under the impression that the user shouldn't be able to change this other than logging in or out, and each log in/out should generate a new BlankNavViewModel.

**Review by navarhontes — CHANGES_REQUESTED** [review]
Made some comments on things that I don't fully understand. Many changes seem to be linting; it's a bit easier to review if linting is done in its own commits. I think it generally looks fine though I frankly don't understand all of it. 

I want to address: is there no need for android tests here?

**Review by candidcalcifer — APPROVED** [review]

File: app/src/main/java/com/courseproject/tindar/BlankNavActivity.java (hunk 1 above)
Comment (jenniferjinyoungyang) [reply app/src/main/java/com/courseproject/tindar/BlankNavActivity.java:32]: This is a view model for Blank Nav activity - so it shares the data between the activity and its fragments. I saw this in google, so I think we cannot share the view model across the activities.

image

File: app/src/main/java/com/courseproject/tindar/BlankNavViewModel.java (hunk 2 above)
Comment (jenniferjinyoungyang) [reply app/src/main/java/com/courseproject/tindar/BlankNavViewModel.java:16]: Honestly, I am not sure, but the examples I found from Google typically work with live data with a view model, and LiveData class itself is abstract so I had to call MutableLivedata. The below link talks about the advantage of using live data.  

https://developer.android.com/topic/libraries/architecture/livedata

**Comment by jenniferjinyoungyang** [comment]
> Made some comments on things that I don't fully understand. Many changes seem to be linting; it's a bit easier to review if linting is done in its own commits. I think it generally looks fine though I frankly don't understand all of it.
> 
> I want to address: is there no need for android tests here?

We are not required to test the frontend for this project, we don't need to add tests for this PR.

**Review by navarhontes — APPROVED** [review]

**Review by kresimirgotovac — APPROVED** [review]

**Pull request merged**